### PR TITLE
Feature/add unit and integration tests to stacked_locator stacked_logger and refactor

### PR DESCRIPTION
**Pull request by eb-so**
- Remove ServicesConfig class and use the list of dependencies directly
- Revert FactoryParameter annotation to FactoryParam
- Setup locator should always have a future<void> return type
- Moved the logic from the stacked_locator_content to each dependency type `registerDependencies` function
- Add integration tests to locator_generator and logger_generator to check for compilation errors
- remove unused route samples
- Breaking change: replace stacked, stacked_annotations imports with stacked_core in locator generator
- Add the stacked_locator and stacked_logger to build.yaml local runner to run them from inside the package
- Remove flutter sdk dependency
- Add dependency_overrides for stacked_core
- Add meta package for visibleForTesting annotation
- Add Unit tests to all the possible scenarios of logger and locator generator content
-  add logger package to the dependency 
- replace kReleaseMode in logger_generator from flutter/foundation with `bool.fromEnvironment("dart.vm.product")` to get rid of flutter dependency

### CODE REVIEW

File: packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_dependency.dart
Comment (FilledStacks) [inline packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_dependency.dart:20]: Nice

File: packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_param_dependency.dart
Comment (FilledStacks) [inline packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_param_dependency.dart]: This can be improved for what you need. I assume you only need to join the list with a `,` in between? If so you can use

```dart
final constructerParamtypes = constructorParamTypes.join(',')
```

File: packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_param_dependency.dart
Comment (FilledStacks) [inline packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_param_dependency.dart]: The same join functionality can be applied here.

File: packages/stacked_generator/lib/src/generators/getit/dependency_config/dependency_config.dart
Comment (FilledStacks) [inline packages/stacked_generator/lib/src/generators/getit/dependency_config/dependency_config.dart:25]: This override abstraction is awesome! Great choice

File: packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart
Comment (FilledStacks) [inline packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart:27]: If we have no presolve then this doesn't need to be a Future. Which is why I had the original check there. 

If it's always a Future that means the main function will always have to be turned into a Future?

File: packages/stacked_generator/test/getit/unit_test/factory_test.dart
Comment (FilledStacks) [inline packages/stacked_generator/test/getit/unit_test/factory_test.dart:23]: Nice set of tests!

**Review by FilledStacks — COMMENTED** [review]
Awesome PR! All the tests are passing. Some failed at first until I ran a dart pub upgrade. 

But now it's all passing. I'm very happy with this PR. It can be merged (after changes) and the same can be applied to other generators where required.

File: packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart
Comment (eb-so) [reply packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart:27]: I think it's more consistent, and there is no harm of doing it

File: packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_param_dependency.dart
Comment (eb-so) [reply packages/stacked_generator/lib/src/generators/getit/dependency_config/factory_param_dependency.dart]: I didn't write this code I just moved it to another file, I will clean it up 👍

File: packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart
Comment (FilledStacks) [reply packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart:27]: @eb-so we'll have to update the stacked readme to direct users to always update the main from `void` to `Future<void>` and call `await setupLocator()` instead of only `setupLocator`

**Review by FilledStacks — APPROVED** [review]
Happy with this PR! Well done, much needed for the project

File: packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart
Comment (eb-so) [reply packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart:27]: I'll make a ticket for that

**Comment by FilledStacks** [comment]
@eb-so let me know what the hold up is of merging this in.

**Comment by eb-so** [comment]
@FilledStacks Nothing, You can merge it 💯

**Pull request merged**
